xbmc/addons/kodi-addon-dev-kit/include/kodi/c-api/addon-instance/pvr/pvr_general.h

```diff
@@ -0,0 +1,288 @@
+  /// passed in the associated @ref kodi::addon::CAddonBase::CreateInstance()
+  /// call.
+  ///
+  /// This makes it possible to select different processing classes within the
+  /// associated add-on.
+  ///
+  ///
+  ///---------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// ...
+  ///
+  /// // On PVR instance of addon
+  /// PVR_ERROR CMyPVRInstance::GetChannelStreamProperties(const kodi::addon::PVRChannel& channel,
+  ///                                                      std::vector<std::pair<std::string, std::string>>& properties)
+  /// {
+  ///   ...
+  ///   // For here on example the inpustream is also inside the PVR addon
+  ///   properties.emplace_back(std::make_pair(PVR_STREAM_PROPERTY_INPUTSTREAM, "pvr.my_one"));
+  ///   properties.emplace_back(std::make_pair(PVR_STREAM_PROPERTY_INPUTSTREAM_INSTANCE_ID, "my_special_id_1"));
+  ///   return PVR_ERROR_NO_ERROR;
+  /// }
+  ///
+  /// ...
+  ///
+  /// // On CAddonBase part of addon
+  /// ADDON_STATUS CMyAddon::CreateInstanceEx(int instanceType,
+  ///                                         std::string instanceID,
+  ///                                         KODI_HANDLE instance,
+  ///                                         KODI_HANDLE& addonInstance
+  ///                                         const std::string& version)
+  /// {
+  ///   if (instanceType == ADDON_INSTANCE_INPUTSTREAM)
+  ///   {
+  ///     kodi::Log(ADDON_LOG_NOTICE, "Creating my special inputstream");
+  ///     if (instanceID == "my_special_id_1")
+  ///       addonInstance = new CMyPVRClientInstance_Type1(instance, version);
+  ///     else if (instanceID == "my_special_id_2")
+  ///       addonInstance = new CMyPVRClientInstance_Type2(instance, version);
+  ///     return ADDON_STATUS_OK;
+  ///   }
+  ///   else if (...)
+  ///   {
+  ///     ...
+  ///   }
+  ///   return ADDON_STATUS_UNKNOWN;
+  /// }
+  ///
+  /// ...
+  /// ~~~~~~~~~~~~~
+  ///
+  #define PVR_STREAM_PROPERTY_INPUTSTREAM_INSTANCE_ID STREAM_PROPERTY_INPUTSTREAM_INSTANCE_ID
+
+  /// @brief the MIME type of the stream that should be played.
+  ///
+  #define PVR_STREAM_PROPERTY_MIMETYPE "mimetype"
+
+  /// @brief <b>"true"</b> to denote that the stream that should be played is a
+  /// realtime stream.
+  ///
+  /// Any other value indicates that this is no realtime stream.
+  ///
+  #define PVR_STREAM_PROPERTY_ISREALTIMESTREAM STREAM_PROPERTY_ISREALTIMESTREAM
+
+  /// @brief <b>"true"</b> to denote that if the stream is from an EPG tag.
+  ///
+  /// It should be played is a live stream. Otherwise if it's a EPG tag it will
+  /// play as normal video.
+  ///
+  #define PVR_STREAM_PROPERTY_EPGPLAYBACKASLIVE "epgplaybackaslive"
+
+  /// @brief Special value for @ref PVR_STREAM_PROPERTY_INPUTSTREAM to use
+  /// ffmpeg to directly play a stream URL.
+  #define PVR_STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG
```

Comment:
Why are the PVR_STREAM_PROPERTY_ defines not in pvr_stream.h ?

Reply:
PVR stream is soon dead and becomes removed.

Reply:
Not sure I understand, butt I trust you.